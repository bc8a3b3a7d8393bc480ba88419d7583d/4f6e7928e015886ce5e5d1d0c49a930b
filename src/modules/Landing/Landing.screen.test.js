import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { testSnapshots } from '../../utils/test.util';
import Container, { LandingScreen as Component } from './Landing.screen';

const mockStore = configureStore();
const componentProps = {
  selectedLanguage: {
    languageCode: 'vi',
    languageName: 'Tiếng Việt',
    countryName: 'Việt Nam',
    countryCode: 'VN',
  },
  setLanguage: jest.fn(),
};

describe('Modules/Landing - Landing Screen - UI component', () => {
  testSnapshots(Component, [
    {
      description: 'basic render',
      props: componentProps,
    },
  ]);
});

describe('Modules/Landing - Landing Screen - Redux Container', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore({
      language: {
        ...componentProps.selectedLanguage,
      },
    });
    store.dispatch = jest.fn();
    wrapper = shallow(<Container store={store} />);
  });

  test('maps state and dispatch to props', () => {
    expect(wrapper.props()).toEqual(
      expect.objectContaining({
        setLanguage: expect.any(Function),
        selectedLanguage: componentProps.selectedLanguage,
      }),
    );
  });
});
