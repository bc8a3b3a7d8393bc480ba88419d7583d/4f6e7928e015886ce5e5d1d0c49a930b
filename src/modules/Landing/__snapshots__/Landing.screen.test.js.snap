// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules/Landing - Landing Screen - UI component snapshots basic render 1`] = `
<LandingView
  onSelectLanguage={[Function]}
  selectedLanguage={
    Object {
      "countryCode": "VN",
      "countryName": "Việt Nam",
      "languageCode": "vi",
      "languageName": "Tiếng Việt",
    }
  }
/>
`;
